File: explorer/normalization/weather/common/paths.mjs
import { dirname, join } from "node:path";
import { fileURLToPath } from "node:url";

const COMMON_DIR = dirname(fileURLToPath(import.meta.url));
export const WEATHER_NORMALIZATION_DIR = join(COMMON_DIR, "..");
export const EXPLORER_ROOT = join(COMMON_DIR, "..", "..", "..");
export const RAW_WEATHER_ROOT = join(EXPLORER_ROOT, "data_raw", "weather");
export const NORMALIZED_WEATHER_ROOT = join(EXPLORER_ROOT, "data_normalized", "weather");
export const PROCESSED_WEATHER_ROOT = join(EXPLORER_ROOT, "data_processed", "weather");
export const WEATHER_FETCH_LOG_PATH = join(RAW_WEATHER_ROOT, "fetch_log.jsonl");

export function normalizedOutputPath(recordFamily, runStamp) {
  return join(NORMALIZED_WEATHER_ROOT, recordFamily, `${recordFamily}_${runStamp}.jsonl`);
}

File: explorer-v2/pipelines/weather/common/paths.py
from __future__ import annotations

from pathlib import Path


EXPLORER_V2_ROOT = Path(__file__).resolve().parents[3]
RAW_WEATHER_ROOT = EXPLORER_V2_ROOT / "data_raw" / "weather"
PROCESSED_WEATHER_ROOT = EXPLORER_V2_ROOT / "data_processed" / "weather"
NORMALIZED_WEATHER_ROOT = EXPLORER_V2_ROOT / "data_normalized" / "weather"
DATABASE_ROOT = EXPLORER_V2_ROOT / "database" / "postgres"
